Source/JavaScriptCore/dfg/DFGFixJumpPadAvailabilityPhase.cpp

```diff
@@ -0,0 +1,93 @@
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+
+#include "config.h"
+#include "DFGFixJumpPadAvailabilityPhase.h"
+
+#if ENABLE(DFG_JIT)
+
+#include "DFGPhase.h"
+
+namespace JSC { namespace DFG {
+
+class FixJumpPadAvailabilityPhase : public Phase {
+public:
+    FixJumpPadAvailabilityPhase(Graph& graph)
+        : Phase(graph, "fix jump pad availability"_s)
+    {
+    }
+
+    bool run()
+    {
+        ASSERT(m_graph.m_form == ThreadedCPS || m_graph.m_form == LoadStore);
+
+        for (BasicBlock* pad : m_graph.m_newJumpPads) {
+            ASSERT(pad->isJumpPad());
+            BasicBlock* successor = pad->successor(0);
+            for (unsigned i = successor->variablesAtHead.size(); i--;) {
+                Node* node = successor->variablesAtHead[i];
+                if (!node)
+                    continue;
+
+                VariableAccessData* variable = node->variableAccessData();
+                Node* phi = m_graph.addNode(Phi, node->origin, OpInfo(variable));
+                pad->phis.append(phi);
+                switch (variable->operand().kind()) {
+                case OperandKind::Argument: {
+                    size_t index = variable->operand().toArgument();
+                    pad->variablesAtHead.atFor<OperandKind::Argument>(index) = phi;
+                    pad->variablesAtTail.atFor<OperandKind::Argument>(index) = phi;
+                    break;
+                }
+                case OperandKind::Local: {
+                    size_t index = variable->operand().toLocal();
+                    pad->variablesAtHead.atFor<OperandKind::Local>(index) = phi;
+                    pad->variablesAtTail.atFor<OperandKind::Local>(index) = phi;
+                    break;
+                }
+                case OperandKind::Tmp: {
+                    size_t index = variable->operand().value();
+                    pad->variablesAtHead.atFor<OperandKind::Tmp>(index) = phi;
+                    pad->variablesAtTail.atFor<OperandKind::Tmp>(index) = phi;
+                    break;
+                }
+                }
+            }
+        }
+
+        m_graph.m_newJumpPads.clear();
+        return true;
+    }
+};
+
+bool performFixJumpPadAvailabilityPhase(Graph& graph)
+{
+    return runPhase<FixJumpPadAvailabilityPhase>(graph);
+}
```

Comment:
So, we will move this to CriticalEdgeBreakingPhase.cpp

Reply:
Removed `performFixJumpPadAvailabilityPhase` and added `performFixJumpPadAvailability` in `DFGCriticalEdgeBreakingPhase.cpp`.